code/lib/cli-storybook/src/automigrate/fixes/addon-globals-api.ts

```diff
@@ -0,0 +1,300 @@
+      viewportsOptions: viewportMigration.options,
+      backgroundsOptions: backgroundsMigration.options,
+    };
+  },
+
+  prompt({ needsViewportMigration, needsBackgroundsMigration }) {
+    return dedent`
+      We've detected that you're using the ${needsViewportMigration && needsBackgroundsMigration ? 'viewport and backgrounds addons' : needsViewportMigration ? 'viewport addon' : 'backgrounds addon'} with the deprecated configuration API.
+      
+      In Storybook 9, ${needsViewportMigration && needsBackgroundsMigration ? 'these addons have' : 'this addon has'} been updated to use the new globals API which ensures a consistent experience while navigating between stories.
+      
+      We'll update your configuration to use the new API:
+      ${needsViewportMigration ? `- ${picocolors.yellow('viewports')} → ${picocolors.yellow('options')} and ${picocolors.yellow('defaultViewport')} → ${picocolors.yellow('initialGlobals.viewport')}\n` : ''}${needsBackgroundsMigration ? `- ${picocolors.yellow('values')} → ${picocolors.yellow('options')} and ${picocolors.yellow('default')} → ${picocolors.yellow('initialGlobals.backgrounds')}\n` : ''}
+      
+      Learn more: ${picocolors.cyan(MIGRATION)}
+    `;
+  },
+
+  async run({ dryRun, result }) {
+    const {
+      previewConfig,
+      needsViewportMigration,
+      needsBackgroundsMigration,
+      viewportsOptions,
+      backgroundsOptions,
+    } = result;
+
+    const getFieldNode = previewConfig.getFieldNode.bind(previewConfig);
+
+    if (needsViewportMigration) {
+      // Get the viewport parameter object
+      const viewports = getFieldNode(['parameters', 'viewport', 'viewports']) as ObjectExpression;
+
+      if (viewportsOptions?.viewports) {
+        // Remove the old viewports property
+        previewConfig.removeField(['parameters', 'viewport', 'viewports']);
+        addProperty(
+          getFieldNode(['parameters', 'viewport']) as ObjectExpression,
+          'options',
+          viewports
+        );
+      }
+
+      // If defaultViewport exists, create initialGlobals.viewport
+      if (viewportsOptions?.defaultViewport) {
+        // Remove the old defaultViewport property
+        const viewportNode = getFieldNode(['parameters', 'viewport']);
+        removeProperty(viewportNode as ObjectExpression, 'defaultViewport');
+
+        previewConfig.setFieldValue(
+          ['initialGlobals', 'viewport', 'value'],
+          viewportsOptions.defaultViewport
+        );
+        previewConfig.setFieldValue(['initialGlobals', 'viewport', 'isRotated'], false);
+      }
+    }
+
+    if (needsBackgroundsMigration) {
+      if (backgroundsOptions?.values) {
+        // Transform values array to options object
+        const optionsObject = transformValuesToOptions(
+          backgroundsOptions.values as ArrayExpression
+        );
+
+        // Remove the old values property
+        previewConfig.removeField(['parameters', 'backgrounds', 'values']);
+        addProperty(
```

Comment:
This could be done differently, by just renaming the key instead of removing everything and re-adding like it's done here:
https://github.com/storybookjs/storybook/blob/728143a09eb6491a520cfbefec675dc52c49aa56/code/lib/cli-storybook/src/automigrate/helpers/addon-a11y-parameters.ts#L68